Upgrade to 3.10 with static pods

smarterclayton opened the pull request:
Before upgrading the control plane, ensure that bootstrapping and other
cluster level operations like installing openshift-sdn and the config sync
plugin have completed. Then upgrade the masters to use bootstrapping mode
one-by-one and have the kubelet take over running the master components as
static pods.

When transitioning each node, move the existing node-config.yaml to
bootstrap-node-config.yaml to allow node config sync to take over if
necessary. Also ensure that the node group config is properly configured
for bootstrapping as v3.9 did not always ensure that config was correct.

Builds on #7486. Has been tested for an existing bootstrapping cluster (GCP) and single node setup.

Things to do:

* [ ] determine any locations where waits are necessary for resiliency
* [ ] wait for SDN to come up cleanly
* [ ] understand why IPAM clearing is safe (breaks masters when openshift-sdn is daemonset), or whether it should be implemented differently
* [ ] etcd backups are failing because it seems to be detecting that we are in system container mode for some reason
* [ ] test with more configurations, including non-bootstrapped
* [ ] warn if incompatible configuration changes are required
* [ ] broken on ansible 2.5, looks like a fact related issue?

smarterclayton commented:
Testing with GCP

```
$ make up WHAT=claytontest REF=release-3.9
$ make upgrade WHAT=claytontest
```

from `openshift/release` in `cluster/test-deploy`.

smarterclayton commented:
Biggest broken behavior right now is openshift-sdn not gracefully running side by side with the installed version.  The SDN (and maybe OVS) pods need to detect whether the kubelet is running with SDN enabled and tolerate it.

vrutkovs commented on roles/openshift_node/tasks/upgrade/config_changes.yml:
```diff
@@ -10,54 +12,111 @@
-# https://docs.openshift.com/container-platform/3.4/admin_guide/overcommit.html#disabling-swap-memory
-# swapoff is a custom module in lib_utils that comments out swap entries in
-# /etc/fstab and runs swapoff -a, if necessary.
-- name: Disable swap
```
Why is it being dropped?

smarterclayton replied:
No cluster that is upgraded should have swap on because we have required swap be on for the last X releases.

vrutkovs commented on roles/openshift_node/tasks/upgrade/config_changes.yml:
```diff
@@ -10,54 +12,111 @@
+      group: root
+      mode: 0600
+  - name: Find existing credentials
+    shell: find "{{ openshift.common.config_base }}/node" -type f -name system\*.kubeconfig
```
Use https://docs.ansible.com/ansible/latest/modules/find_module.html instead

mrogers950 commented on playbooks/common/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml:
```diff
@@ -45,6 +46,56 @@
 
+- import_playbook: label_nodes.yml
+
+# To upgrade, we need masters to be capable of signing certificates
```
If a cluster is using an external CA (no key to sign) and upgrades with the default of bootstrapping enabled, will we abort? For the external CA use case we will need to make sure that bootstrapping is off until we have the pieces in place to support that..

smarterclayton replied:
Yeah, it's in my check list up at the top.

smarterclayton commented:
This contains HA fixes for multi-master bringup.  Anyone hitting issues deploying should try this PR.

At this point I think all node labelling issues are fixed in both bootstrapping and non-bootstrapping modes.

michaelgugino commented on playbooks/common/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml:
```diff
@@ -45,6 +46,59 @@
+# TODO: need to verify settings about the bootstrap configs
+# 1. Does network policy match the master config
+
+- hosts: oo_all_hosts
```
Don't do anything against oo_all_hosts during control plane upgrades.  Don't do anything against nodes during control plane upgrades.

smarterclayton replied:
We’re changing a control plane dynamic component that affects all nodes (like daemonsets).  This particular step may no longer be necessary since I added a ratchet to dynamic config, which I’ll verify.

smarterclayton replied:
I'll test without it, I agree in general although the model with daemonsets won't really match that.  Control plane components will upgrade, and the components will have to tolerate older nodes.  We obviously don't want to disrupt workloads, but upgrading the control plane will affect nodes (sdn, registry, router, config sync, bootstrapping etc all have impacts).

This was intended to pause dynamic reconfiguration until nodes reestablished it.

michaelgugino commented on playbooks/common/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml:
```diff
@@ -45,6 +46,59 @@
   - set_fact:
       pre_upgrade_complete: True
 
+- import_playbook: label_nodes.yml
```
We shouldn't need to label nodes here.  Everything should have been labeled properly during 3.9 upgrades and during 3.9 / 3.10 installs.

smarterclayton replied:
No, 3.10 changes a default to make infra nodes correct.  3.9 doesn’t have the correct labels.

mtnbikenc commented:
Testing mulit-master ocp install with this PR.

michaelgugino commented on playbooks/common/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml:
```diff
@@ -45,6 +46,59 @@
+  - role: openshift_sdn
+    when: openshift_use_openshift_sdn | default(True) | bool
+
+- name: Update master nodes
```
Are we sure we want to upgrade nodes on the masters before upgrade the control plane?

smarterclayton replied:
For 3.10 yes, we’ll just require that 3.10 nodes work with a 3.9 master.  In future releases this is less of an issue.  What specifically are you concerned with?

michaelgugino replied:
I just wanted to make sure this code landed in the right spot.  It's the reverse of what we were doing previously.

michaelgugino commented on roles/etcd/tasks/backup/backup.yml:
```diff
@@ -49,10 +49,17 @@
   - "'svirt_sandbox_file_t' not in l_etcd_selinux_labels.stdout"
 
-- name: Generate etcd backup
+- name: Generate etcd backup (legacy)
```
In what context will this legacy backup be used from the 3.10+ branch?  Is this still needed?

smarterclayton replied:
It’s needed for 3.9->3.10, we won’t have a static pod until after the control plane updates.

michaelgugino commented on roles/openshift_node/tasks/upgrade/config_changes.yml:
```diff
@@ -1,7 +1,9 @@
-- name: Update systemd units
-  import_tasks: ../systemd_units.yml
-  when: openshift_is_containerized | bool
+- name: Ensure the node static pod directory exists
```
It would be nice if we could break this file apart into component areas.  It's just really large as is and it looks like we can probably optimize the ordering of some things to group them into logical units.

smarterclayton replied:
Yes

smarterclayton replied:
Slightly better factored now

michaelgugino commented on roles/openshift_node/tasks/upgrade/config_changes.yml:
```diff
@@ -10,54 +12,134 @@
-  swapoff: {}
-  when: openshift_disable_swap | default(true) | bool
+- name: Uninstall openvswitch
+  package:
```
This will fail for atomic?

michaelgugino commented on roles/openshift_node/tasks/upgrade/config_changes.yml:
```diff
@@ -10,54 +12,134 @@
-    key: "{{ item.key }}"
-    value: "{{ item.value }}"
+- name: Remove old service information
+  file: path={{ item }} state=absent
```
Use dictionary format instead of x=y

michaelgugino commented on roles/openshift_node/tasks/upgrade/config_changes.yml:
```diff
@@ -10,54 +12,134 @@
-    src: "node.service.j2"
-  register: l_node_unit
-  when: not openshift_is_containerized | bool
+- name: Move existing credentials and configuration into bootstrap configuration
```
Let's make this block it's own file instead.

smarterclayton replied:
Ok

michaelgugino commented on roles/openshift_node/tasks/upgrade/config_changes.yml:
```diff
@@ -10,54 +12,134 @@
+    copy:
+      remote_src: true
+      src: "{{ openshift.common.config_base }}/node/node-config.yaml"
+      dest: "{{ openshift.common.config_base }}/node/bootstrap-node-config.yaml"
```
This needs to have some logic to ensure it only happens just the once.  Don't want to forget about this for future upgrades and someone's old, stale config file stomps the newer one.

This would also affect minor upgrades (eg 3.10.0 -> 3.10.1)

michaelgugino replied:
Seems we're deleting the old file in later step.  Would need to account for the origin src being absent.

smarterclayton replied:
Yes, this should only happen if bootstrap doesn’t exist and the source does.

michaelgugino commented on roles/openshift_node/tasks/upgrade/config_changes.yml:
```diff
@@ -10,54 +12,134 @@
+    copy:
+      remote_src: true
+      src: "{{ item }}"
+      dest: "{{ openshift.common.config_base }}/node/bootstrap.kubeconfig"
```
Same comment as above, probably only want to do this once.

michaelgugino commented on roles/openshift_node/tasks/upgrade_pre.yml:
```diff
@@ -5,6 +5,18 @@
 - set_fact:
     skip_node_svc_handlers: True
 
+- name: Wait for node to be ready
```
Why are we waiting here?  This is the pre_upgrade step.  Nothing should have been restarted at this point.

smarterclayton replied:
An unready node can’t be safely upgraded when bootstrapping is in play.  I think if this as a precondition rather than a failure mode.

smarterclayton replied:
Will verify this is still needed.

michaelgugino commented on roles/openshift_node_group/files/sync.yaml:
```diff
@@ -71,12 +64,49 @@ spec:
             touch /tmp/.old
           fi
 
+          # loop until BOOTSTRAP_CONFIG_NAME is set
```
Ship this as part of some image?  This seems crufty to have a giant bash script embedded like this.

smarterclayton replied:
Maybe later.  Config is config.  Image is not any easier to manage.

smarterclayton replied:
In this case sync belongs to how we are configuring dynamic sync during install.  This is serverless bash

smarterclayton replied:
Note that I will probably collapse a lot of `sdn.yaml` into the networking go binary once we get the transition locked down so it's stable.

In the future all of the sync code will be moving into the kubelet or another component, I just don't have that component now and I want to get the algorithm correct for now.

smarterclayton commented:
Looks like the nodename thing came in when I rebased, testing a fix.

smarterclayton commented:
Seeing a new failure on re-run against an HA cluster that was partially installed:

```
Failure summary:


  1. Hosts:    ci-clayton-ha-2-ig-m-btgj
     Play:     Inspect state of first master config settings
     Task:     Gather existing session secrets from first master
     Message:  The conditional check 'l_existing_osm_session.secrets is defined' failed. The error was: error while evaluating conditional (l_existing_osm_session.secrets is defined): {{ (osm_session_secrets_stat.content | b64decode | from_yaml) }}: 'osm_session_secrets_stat' is undefined

               The error appears to have been in '/usr/share/ansible/openshift-ansible/roles/openshift_control_plane/tasks/generate_session_secrets.yml': line 18, column 3, but may
               be elsewhere in the file depending on the exact syntax problem.

               The offending line appears to be:

               # roles/lib_utils/filter_plugins/oo_filters.py
               - name: Gather existing session secrets from first master
                 ^ here
```

Looks unrelated to changes I made?

smarterclayton commented:
Made some changes to how the node playbooks are organized:

1. unified the three facts related to node bootstrapping:
  1. `openshift_node_bootstrap` inventory variable that defines whether a node should be bootstrapped, defaults to true, masters are always bootstrapped
  2. `openshift_is_bootstrapped` inventory variable that can be set if the node is known to have already been prepped for bootstrap, like a golden image (for dynamic node groups), or a basic_fact discovered during init if `/etc/origin/node/bootstrap-node-config.yaml` exists.
1. added oo_nodes_to_bootstrap group which is a subset of oo_nodes_to_config for all nodes which are either already bootstrapped or are `openshift_node_bootstrap=true`
1. two primary node configuration entrypoints, bootstrap.yml (called first, does the equivalent of image_prep, operates on oo_nodes_to_bootstrap), and config.yml (called second, same as before, operates on oo_nodes_to_config:!oo_nodes_to_bootstrap).
  1. Reorganized the plays between the two to share, use `l_node_group` to parameterize which group to target

Also folded in the bug I hit above (I'll rebase onto whatever the different setup is), and addressed a chunk of the round of comments.  I'm continuing to refine the transition from openvswitch on node to openvswitch in daemonset in a way that minimizes downtime.

vrutkovs commented:
Several weird errors from 3 master setup on Atomic:
[ocp-master3 logs](https://s3.amazonaws.com/aos-ci/ghprb/openshift/openshift-ansible/e29f6e8fcaa4d337aa1db7155f289db39e21b665.1.1523009107748838094/artifacts/journals/ocp-master3.log):
```
rbac.go:116] RBAC DENY: user "system:node:ocp-master2" groups ["system:nodes" "system:authenticated"] cannot "create" resource "certificatesigningrequests.certificates.k8s.io/selfnodeclient" cluster-wide
```
```
transport.go:108] The currently active client certificate has expired, but the server is not responsive. A restart may be necessary to retrieve new initial credentials.
```
```
kubelet_node_status.go:406] Error updating node status, will retry: error getting node "ocp-master3": Unauthorized
```

mtnbikenc commented:
Tested multi-master ocp install and the installer completed.  However, the kube-service-catalog controller-managers were in CrashLoopBackOff.

mtnbikenc commented:
```
[ec2-user@ip-172-18-14-48 ~]$ oc logs po/controller-manager-2msg7 -n kube-service-catalog
Flag --port has been deprecated, see --secure-port instead
I0406 14:45:03.026215       1 feature_gate.go:184] feature gates: map[OriginatingIdentity:true]
I0406 14:45:03.029454       1 feature_gate.go:184] feature gates: map[OriginatingIdentity:true AsyncBindingOperations:true]
I0406 14:45:03.029497       1 hyperkube.go:188] Service Catalog version v0.0.0-master+$Format:%h$ (built 2018-04-04T13:43:16Z)
W0406 14:45:03.029528       1 controller_manager.go:97] program option --port is obsolete and ignored, specify --secure-port instead
I0406 14:45:03.029541       1 controller_manager.go:101] Building k8s kubeconfig
I0406 14:45:03.031559       1 controller_manager.go:129] Building service-catalog kubeconfig for url:
I0406 14:45:03.031574       1 controller_manager.go:136] Using inClusterConfig to talk to service catalog API server -- make sure your API server is registered with the aggregator
Error: failed to establish SecureServingOptions error reading /var/run/kubernetes-service-catalog/apiserver.key, certificate and key must be supplied as a pair
```

mtnbikenc commented:
Also noted that the compute nodes are NoteReady.
```
[ec2-user@ip-172-18-14-48 ~]$ oc get nodes
NAME                            STATUS     ROLES          AGE       VERSION
ip-172-18-0-231.ec2.internal    Ready      infra,master   1h        v1.9.1+a0ce1bc657
ip-172-18-14-48.ec2.internal    Ready      infra,master   1h        v1.9.1+a0ce1bc657
ip-172-18-15-238.ec2.internal   Ready      infra,master   1h        v1.9.1+a0ce1bc657
ip-172-18-2-41.ec2.internal     NotReady   compute        1h        v1.9.1+a0ce1bc657
ip-172-18-3-100.ec2.internal    NotReady   compute        1h        v1.9.1+a0ce1bc657
```

```
KubeletNotReady              runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
```

mtnbikenc commented:
Tested with commit 2832ec3.  Same issues as before.
* Compute nodes are NotReady
* ASB deploy is pending
* kube-service-catalog controller-manager pods are CrashLoopBackOff

smarterclayton commented:
/test install

smarterclayton commented:
Debugging the single node case.  Identified an issue with openshift-sdn
where networks were lost, that should be addressed in my next push.

On Fri, Apr 6, 2018 at 3:11 PM, Russell Teague <notifications@github.com>
wrote:

> Tested with commit 2832ec3
> <https://github.com/openshift/openshift-ansible/commit/2832ec36e21675a24cd0604f2da1d293300e581b>.
> Same issues as before.
>
>    - Compute nodes are NotReady
>    - ASB deploy is pending
>    - kube-service-catalog controller-manager pods are CrashLoopBackOff
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-ansible/pull/7723#issuecomment-379349625>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p-RfHb6Z08Plp-v6tTWkBUX_SgQuks5tl73egaJpZM4TCujn>
> .
>

smarterclayton commented:
install failure on previous run was a hang in the sync pod - i *think* it's networking related ,but still testing

smarterclayton commented:
Ok, I was able to do both ha and non HA upgrade runs (one for bootstrapping and one without).  Still working through the checklist but this is close.  The install changes need to be merged, but splitting them out is likely to introduce more iteration.  Any reason not to merge with the review comments and iterate after?

smarterclayton commented:
/retest

smarterclayton commented:
I think I have all the feedback addressed (still the list of things at the top).

smarterclayton commented:
/test all

smarterclayton commented:
/retest

vrutkovs commented:
Now HA install fails on FAW as `/var/lib/kubeletet` is not mounted on the host (see [ocp-master1 logs](https://s3.amazonaws.com/aos-ci/ghprb/openshift/openshift-ansible/7ba4ff96204c16490e2acff9606c69cabff87946.1.1523277333463852391/artifacts/journals/ocp-master1.log))

Upgrade fails here at playbooks/common/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml - `- meta: clear_facts` would remove all cluster facts and should be disabled. After removing this line the upgrade breaks, as `origin-node` service was stopped and no API/controller containers were running - looking at details of that

mtnbikenc commented:
Tested HA OCP install.  Install completed however compute nodes are NotReady. ASB deploy pod is `Pending`.

```
NAME                           STATUS     ROLES          AGE       VERSION
ip-172-18-2-135.ec2.internal   Ready      infra,master   23m       v1.9.1+a0ce1bc657
ip-172-18-2-22.ec2.internal    Ready      infra,master   23m       v1.9.1+a0ce1bc657
ip-172-18-3-214.ec2.internal   NotReady   compute        17m       v1.9.1+a0ce1bc657
ip-172-18-5-161.ec2.internal   NotReady   compute        17m       v1.9.1+a0ce1bc657
ip-172-18-5-30.ec2.internal    Ready      infra,master   23m       v1.9.1+a0ce1bc657
```
```
  Ready            False   Mon, 09 Apr 2018 13:58:45 +0000   Mon, 09 Apr 2018 13:39:13 +0000   KubeletNotReady              runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
```

mtnbikenc commented:
On compute nodes:
```
[ec2-user@ip-172-18-5-30 ~]$ oc logs po/sdn-qr6rn
2018/04/09 14:32:44 socat[6] E connect(5, AF=1 "/var/run/openshift-sdn/cni-server.sock", 40): No such file or directory
warning: Cannot find existing node-config.yaml, waiting 15s ...
```

mtnbikenc commented:
The compute node has /etc/origin/node/bootstrap-node-config.yaml

smarterclayton commented:
The atomic thing is almost certainly that the system container for docker isn't bind mounting /var/lib/kubelet r/w, but could be related.

smarterclayton commented:
Debugging with russell, I think boolean logic in bootstrap_config.yml is screwed up (i want to set the line if it is missing, present and empty, or if force is set).  I think it's only getting set if it's empty.

mtnbikenc commented:
I successfully installed an HA OCP cluster.  All previous issues have been addressed.  There is an issue with the ASB pulling from the wrong registry but I believe that is an unrelated bug.  I'll test the Upgrade playbook against this cluster for basic validation.

mtnbikenc commented:
Upgrade fails:

```
TASK [Run health checks (upgrade)] ***********************************************************************************************************
task path: /home/rteague/git/clusters/aws-c2/openshift-ansible/playbooks/common/openshift-cluster/upgrades/pre/config.yml:34
Monday 09 April 2018  14:51:53 -0400 (0:00:00.205)       0:01:09.843 ********** 
fatal: [ec2-52-201-84-123.compute-1.amazonaws.com]: FAILED! => {"changed": false, "msg": "'openshift' is undefined, did 'openshift_facts' run?", "playbook_context": "upgrade"}
fatal: [ec2-54-175-8-201.compute-1.amazonaws.com]: FAILED! => {"changed": false, "msg": "'openshift' is undefined, did 'openshift_facts' run?", "playbook_context": "upgrade"}
fatal: [ec2-34-229-187-17.compute-1.amazonaws.com]: FAILED! => {"changed": false, "msg": "'openshift' is undefined, did 'openshift_facts' run?", "playbook_context": "upgrade"}
fatal: [ec2-204-236-249-154.compute-1.amazonaws.com]: FAILED! => {"changed": false, "msg": "'openshift' is undefined, did 'openshift_facts' run?", "playbook_context": "upgrade"}
```

mtnbikenc commented:
@sdodson @michaelgugino 
This task needs to go.  https://github.com/openshift/openshift-ansible/blob/master/playbooks/common/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml#L22

It kills openshift facts in 2.5.  There might be an update to how that task works in Ansible 2.5.

Reproducing this on 3.10 upgrades.

mtnbikenc commented:
Opened #7869 to fix upgrades.

mtnbikenc commented:
With the `clear_facts` task removed, the upgrade fails at this task:
```
TASK [etcd : Generate etcd backup (legacy)] **************************************************************************************************
task path: /home/rteague/git/clusters/aws-c2/openshift-ansible/roles/etcd/tasks/backup/backup.yml:52
Monday 09 April 2018  16:26:25 -0400 (0:00:00.473)       0:07:21.908 ********** 
fatal: [ec2-52-201-84-123.compute-1.amazonaws.com]: FAILED! => {"changed": false, "cmd": "etcdctl backup --data-dir=/var/lib/etcd/ --backup-dir=/var/lib/etcd//openshift-backup-pre-upgrade-20180409162621", "msg": "[Errno 2] No such file or directory", "rc": 2}
fatal: [ec2-54-175-8-201.compute-1.amazonaws.com]: FAILED! => {"changed": false, "cmd": "etcdctl backup --data-dir=/var/lib/etcd/ --backup-dir=/var/lib/etcd//openshift-backup-pre-upgrade-20180409162621", "msg": "[Errno 2] No such file or directory", "rc": 2}
fatal: [ec2-34-229-187-17.compute-1.amazonaws.com]: FAILED! => {"changed": false, "cmd": "etcdctl backup --data-dir=/var/lib/etcd/ --backup-dir=/var/lib/etcd//openshift-backup-pre-upgrade-20180409162621", "msg": "[Errno 2] No such file or directory", "rc": 2}
```

smarterclayton commented:
I didn’t see the backup thing, can you verify you have /var/lib/etcd?  Note that all of my upgrade runs are on 2.4 and we are not supporting 2.5 for 3.10 so if this is another 2.5 issue I don’t want to rathole.

smarterclayton commented:
/test gcp

mtnbikenc commented:
I started a new install/upgrade with 2.4.3.

giuseppe commented:
@smarterclayton @sjenning @giuseppe @vrutkovs, I've opened a PR here: https://github.com/openshift/origin/pull/19284

mtnbikenc commented:
Using this PR (rebased to master) I performed a 3.10 install followed by running an upgrade.
3 nodes master/infra/etcd
2 nodes compute
1 node lb

smarterclayton commented:
CI related failure.

dobbymoodge commented:
/refresh

mtnbikenc commented on playbooks/openshift-master/private/tasks/enable_bootstrap_config.yml:
```diff
@@ -0,0 +1,10 @@
+  import_role:
+    name: openshift_bootstrap_autoapprover
+  when:
+  - openshift_master_bootstrap_auto_approve | default(False) | bool
```
Should this default to True?  Trying to understand the process and if this is the default behavior.  My current install requires openshift_master_bootstrap_auto_approve=True in inventory.

smarterclayton replied:
If you turn it off what happens?  I have successful installs without it, because the playbooks run CSR themselves.  If you're seeing failures it may be other inventory flags?

mtnbikenc replied:
If I don't use `openshift_master_bootstrap_auto_approve=True`, the compute nodes fail, but the playbook continues and processes the entire rest of the install and then reports a failure at the end.  Upon checking cluster status, everything is up as expected.


```
Failure summary:


  1. Hosts:    ec2-34-235-156-129.compute-1.amazonaws.com, ec2-54-227-73-138.compute-1.amazonaws.com
     Play:     Distribute bootstrap and start nodes
     Task:     openshift_node : fail
     Message:  Node start failed.
```

The task where the nodes actually fail is here:
```
TASK [openshift_node : Start and enable node for bootstrapping]
```
http://file.rdu.redhat.com/~rteague/temp/2018-04-10T15.18.40-deploy.log

dobbymoodge commented:
/retest one more time

smarterclayton commented:
Why is red hat content being included in a Centos image?

sdodson commented:
> #7878 was the break. I also merged openshift/origin#19284 and I'm waiting on https://ci.openshift.redhat.com/jenkins/job/push_origin_release/2705/ to complete. When that is this should be green - do we want to hold or merge and iterate in a followup? QE is blocked without this.

We discussed this during standup today. The opinion of the team is that we should go ahead and merge this to stem the tide of incoming bugs associated with fixes that exist already in this PR. We want to produce a prioritized list of supported upgrade paths so that we can focus CL and QE efforts in the right areas.

smarterclayton commented:
/retest

smarterclayton commented:
Here's the gut of the CI job I am creating:

1. GCP, 3.9 install using 3 masters, 1 infra, 1 node with using the default label selectors, no bootstrapping
2. Upgrades to 3.9 bootstrapped, but not autoscaling groups
3. Run a basic smoke test setup.

I'm going to try to make inventory easily accessible so people can fork it themselves.

smarterclayton commented:
bot, retest this please

vrutkovs commented:
`container init caused \\\"rootfs_linux.go:58: mounting \\\\\\\"/var/lib/kubelet\\\\\\\" to rootfs \\\\\\\"/var/lib/containers/atomic/origin-node.0/rootfs\\\\\\\" at \\\\\\\"/var/lib/kubelet\\\\\\\" caused \\\\\\\"stat /var/lib/kubelet: no such file or directory\\\\\\\"
`
It seems the folder needs to be created first - see https://github.com/openshift/openshift-ansible/pull/7872/commits/a6a8014bef0ebdb591bfcc4178aec875b51a89a8#diff-76f227d5091c1c4e68b62433eedb879eR15

sdodson commented:
bot, retest this please

smarterclayton commented:
Pushed an ansible fix for atomic.

smarterclayton commented:
/retest

smarterclayton commented:
/test gcp-upgrade

Trying out the new GCP HA upgrade from 3.9 to 3.10 job (uses a static config 'gcp-ha-static' for 3.9, then upgrades to 'gcp-ha' for 3.10).

smarterclayton commented:
/refresh

smarterclayton commented:
/test gcp

smarterclayton commented:
/test tcp-upgrade

sdodson commented:
/test gcp-upgrade

smarterclayton commented:
Failure was a test flake, merging in a few minutes once I figure out why it didn't post a junit result.
